Client/lib/gMap.js

```diff
@@ -0,0 +1,231 @@
+    
+    //----------------------------------
+    //styling the Map
+    //----------------------------------
+
+    var mapStyles = [
+      {
+        stylers: [ { hue: "#00ffe6" }, { saturation: -20 } ]
+      },{
+        featureType: "road",
+        elementType: "geometry",
+        stylers: [ { lightness: 100 }, { visibility: "simplified" } ]
+      },{
+        featureType: "road",
+        elementType: "labels",
+        stylers: [ { visibility: "off" } ]
+      }
+    ];
+
+    var styledMap = new google.maps.StyledMapType(mapStyles,{name: "Styled Map"});
+    map.mapTypes.set('map_style', styledMap);
+    map.setMapTypeId('map_style');
+
+    //----------------------------------
+    //events
+    //----------------------------------
+    google.maps.event.addListener(map, 'click', function(event) {
+      addMarker(event.latLng, 'Count ' + markerCords.length);
+    });
+
+    //----------------------------------
+    //adding custom GUI elements
+    //----------------------------------
+
+
+    var tagForZoomOut = document.createElement('div');
+    var controlForZoomOut = new CenterControl(tagForZoomOut, '-', function(){
+      map.setZoom(map.getZoom()-1);
+    });
+
+    var tagForZoomIn = document.createElement('div');
+    var controlForZoomIn = new CenterControl(tagForZoomIn, '+', function(){
+      map.setZoom(map.getZoom()+1);
+    });
+
+    var tagForWalk = document.createElement('div');
+    var controlForWalk = new CenterControl(tagForWalk, 'W', function(){
+      travelMode = 'WALKING';
+    },true,true);
+    
+    var tagForBicycle = document.createElement('div');
+    var controlForBicycle = new CenterControl(tagForBicycle, 'B', function(){
+      travelMode= 'BICYCLING';
+    },true);
+    
+    var tagForDrive = document.createElement('div');
+    var controlForDrive = new CenterControl(tagForDrive, 'D', function(){
+      travelMode= 'DRIVING';
+    },true);
+
+    var tagForTransit = document.createElement('div');
+    var controlForTransit = new CenterControl(tagForTransit, 'T', function(){
+      travelMode= 'TRANSIT';
+    },true);
+
+    //adding the events tags to the Map
+    
+    map.controls[google.maps.ControlPosition.TOP_LEFT].push(tagForZoomOut);
+    map.controls[google.maps.ControlPosition.TOP_LEFT].push(tagForZoomIn);
+    map.controls[google.maps.ControlPosition.BOTTOM_LEFT].push(tagForWalk);
+    map.controls[google.maps.ControlPosition.BOTTOM_LEFT].push(tagForBicycle);
+    map.controls[google.maps.ControlPosition.BOTTOM_LEFT].push(tagForDrive);
+    map.controls[google.maps.ControlPosition.BOTTOM_LEFT].push(tagForTransit);
+  }
+
+
+
+
+  //------------------------
+  //functions
+  //------------------------
+
+  function calcRoute() {
+
+    var directionsDisplay = new google.maps.DirectionsRenderer({suppressMarkers: true});
+    var directionsService = new google.maps.DirectionsService();
+  
+    directionsDisplay.setMap(map);
+
+    var start = markerCords[markerCords.length-2];
```

Comment:
This entire functions has been removed and replaced with createPath with that change markerCords has been deleted.